lib/src/bar-chart/BarChart.tsx

```diff
@@ -0,0 +1,179 @@
+import React, { memo } from "react";
+import { BarChart } from "@cloudscape-design/components";
+import { applyTheme } from "@cloudscape-design/components/theming";
+import { useState, useMemo, useCallback } from "react";
+import BarChartProps, { DataTypes, InsetWrapperProps } from "./types";
+import theme from "./theme";
+import styled, { css } from "styled-components";
+import { DxcSpinner, DxcInset, DxcSelect, DxcButton, DxcHeading, DxcGrid } from "../main";
+import DxcIcon from "../icon/Icon";
+import CoreTokens from "../common/coreTokens";
+import useTranslatedLabels from "../useTranslatedLabels";
+
+/**
+ * Cloudscape Design Components - Theming
+ * https://cloudscape.design/foundation/visual-foundation/theming/
+ */
+applyTheme({ theme });
+
+const StatusMessage = memo(({ title, description }: { title: string; description: string }) => (
+  <StatusMessageContainer>
+    <strong>{title}</strong>
+    {description}
+  </StatusMessageContainer>
+));
+
+const InsetWrapper = ({ condition, children }: InsetWrapperProps): JSX.Element => (
+  <>{condition ? <DxcInset top="0.5rem">{children}</DxcInset> : children}</>
+);
+
+const DxcBarChart = <X extends DataTypes>({
+  chartTitle,
+  error,
+  horizontalBars,
+  legendTitle,
+  loading,
+  onFilterChange,
+  onHighlightChange,
+  onRetry,
+  series,
+  showFilter,
+  showLegend,
+  stackedBars,
+  xDomain,
+  xTitle,
+  yDomain,
+  yTitle,
+}: BarChartProps<X>) => {
+  const [filteredData, setFilteredData] = useState(series);
+  const filters = useMemo(() => series.map((series) => series.title), [series]);
+  const [selectedFilters, setSelectedFilters] = useState<string[]>(filters);
+  const translatedLabels = useTranslatedLabels();
+
+  const filterData = useCallback(
+    (value: string[]) => {
+      setSelectedFilters(value);
+      setFilteredData(series.filter((series) => value.includes(series.title)));
+    },
+    [series]
+  );
+
+  return loading ? (
+    <LoadingContainer>
+      <DxcSpinner mode="small" />
+      {translatedLabels.barChart.loadingDataMessage}
+    </LoadingContainer>
+  ) : error ? (
+    <ErrorStatus>
+      <Label>
+        <DxcIcon icon="cancel" />
+        {error}
+      </Label>
+      <DxcButton
+        label={translatedLabels.barChart.retryButtonLabel}
+        mode="secondary"
+        onClick={() => {
+          onRetry();
```

Comment:
While most functional tests are unnecessary. I do think we should add some test for how the bar chart interacts with our components.

Reply:
Yes, I agree on this but there's a problem with Cloudscape's Bar Chart and our Jest configurations I couldn't solve...

Reply:
I'm giving it a second try.

Reply:
Something that I did not consider is that a condition can be added to check if onRetry is available before rendering the button since it is not a mandatory prop.

Reply:
Updated. I also think it should be optional.